src/EditorFeatures/CSharpTest/ConvertAutoPropertyToFullProperty/ConvertAutoPropertyToFullPropertyTests.cs

```diff
@@ -0,0 +1,985 @@
+        {
+            _goo = value;
+        }
+    }
+
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task WithInitialValue()
+        {
+            var text = @"
+class goo
+{
+    public int G[||]oo { get; set; } = 2
+}
+";
+            var expected = @"
+class goo
+{
+    private int _goo = 2;
+
+    public int Goo
+    {
+        get
+        {
+            return _goo;
+        }
+        set
+        {
+            _goo = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task WithCalculatedInitialValue()
+        {
+            var text = @"
+class goo
+{
+    const int num = 345;
+    public int G[||]oo { get; set; } = 2*num
+}
+";
+            var expected = @"
+class goo
+{
+    const int num = 345;
+    private int _goo = 2 * num;
+
+    public int Goo
+    {
+        get
+        {
+            return _goo;
+        }
+        set
+        {
+            _goo = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task WithPrivateSetter()
+        {
+            var text = @"
+class goo
+{
+    public int G[||]oo { get; private set; }
+}
+";
+            var expected = @"
+class goo
+{
+    private int _goo;
+
+    public int Goo
+    {
+        get
+        {
+            return _goo;
+        }
+        private set
+        {
+            _goo = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task WithFieldNameAlreadyUsed()
+        {
+            var text = @"
+class goo
+{
+    private int _goo;
+
+    public int G[||]oo { get; private set; }
+}
+";
+        var expected = @"
+class goo
+{
+    private int _goo;
+    private int _goo1;
+
+    public int Goo
+    {
+        get
+        {
+            return _goo1;
+        }
+        private set
+        {
+            _goo1 = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task WithComments()
+        {
+            var text = @"
+class goo
+{
+    // Comments before
+    public int G[||]oo { get; private set; } //Comments during
```

Comment:
Worth testing with trivia on the `get` or `set`?

Reply:
I did some ad hoc testing on this and it works fine with trivia before and after both get and set.